### DISCUSSION TITLE
CosmosOS not building ISO with File System

### DISCUSSION THREAD

**Discussion by SmashedFrenzy16** [CosmosOS/Cosmos#2872] [Q&A]
**Exception**
![Screenshot 2023-12-24 132710](https://github.com/CosmosOS/Cosmos/assets/68993968/03916e58-9f61-417e-bc7d-d8c7e537622c)

NOTE: I got this when running the ISO on VMWare on Windows, and VirtualBox on Mac.

**Visual Studio Output Logs**

![Screenshot 2023-12-24 132728](https://github.com/CosmosOS/Cosmos/assets/68993968/bc239f66-85e3-47af-a226-e2a20960753a)
![Screenshot 2023-12-24 132739](https://github.com/CosmosOS/Cosmos/assets/68993968/bd59a1d2-0031-44be-a261-58c74f76744d)

NOTE: The second error is obtained after pressiong "Ignore". "Retry" just launches a Visual Studio debugger, which has not done anything to improve my situation (it doesn't seem to do anything). These errors also don't affect anything that isn't the file system code, but maybe they are a part of the problem.

**How To Reproduce**
In the project properties, I disabled remote debugging and set the launch profile to ISO

**Context**

When i had utilised the file system in my code, I could not get it to work on the ISO, as I got the kernel error. The code works without those specific commands though. I then found that I had to format the drive to make it work on the ISO, but I could not find a way to actually do so. I can only run the file system using the VMware launch option, but I need it to work on the ISO as well. These are the commands I used for the file system so far: 

```
Sys.FileSystem.CosmosVFS fileSys = new Cosmos.System.FileSystem.CosmosVFS();

Sys.FileSystem.VFS.VFSManager.RegisterVFS(fileSys);
```

The second command is inside the `BeforeRun()` function, and the first is before.

**Comment by randomusert** [comment]
It worked for me. [check](https://github.com/tinteeam/tinos_version2) the repo
